boa_engine/src/syntax/ast/node/mod.rs

```diff
@@ -348,6 +349,92 @@ where
     buf
 }
 
+/// `FormalParameterList` is a list of `FormalParameter`s that describes the parameters of a function.
+///
+/// More information:
+///  - [ECMAScript reference][spec]
+///
+/// [spec]: https://tc39.es/ecma262/#prod-FormalParameterList
+#[cfg_attr(feature = "deser", derive(Serialize, Deserialize))]
+#[derive(Clone, Debug, Default, PartialEq, Trace, Finalize)]
+pub struct FormalParameterList {
+    pub(crate) parameters: Box<[FormalParameter]>,
+    #[unsafe_ignore_trace]
+    pub(crate) flags: FormalParameterListFlags,
+}
+
+#[allow(clippy::fn_params_excessive_bools)]
+impl FormalParameterList {
+    pub(crate) fn new(
+        parameters: Box<[FormalParameter]>,
+        is_simple: bool,
+        has_duplicates: bool,
+        has_rest_parameter: bool,
+        has_expressions: bool,
+        has_arguments: bool,
```

Comment:
Here I would prefer to either pass the flags directly or have some kind of builder pattern. Actually, passing the flags is as simple as writing `IS_SIMPLE | HAS_DUPLICATES`, for example. And bitflags also has some functional approaches to do these things cleanly. That way we can remove the clippy "allow".

Also, it would be nice to have some minimal documentation in the methods, even if it's mostly clear what they do.